encrypter.cpp

```diff
@@ -0,0 +1,177 @@
+#include <iostream>
+#include <string>
+#include <fstream>
+#include <ios>
+#include <random>
+#include <sstream>
+#include <utility>
+#include <iterator>
+
+using Byte = char;
+
+//=====put_error=====//
+void put_error(const std::string& error_message)
+{
+  std::cout << error_message;
+  exit(1);
+}
+
+//=====shuffle=====//
+void shuffle(std::vector<Byte>& input)
+{
+  std::default_random_engine gen;
+
+  for (size_t i = 0; i < input.size(); ++i)
+  {
+    std::uniform_int_distribution<int> dis (i, input.size() - 1);
+    std::swap(input[i], input[dis(gen)]);
+  } 
+}
+
+//=====write_file=====//
+void write_file(const std::string& filename, std::vector<Byte> output, const std::string error_message)
+{
+  std::ofstream file (filename, std::ios::out|std::ios::binary|std::ios::trunc);
+  if (!file.is_open())
+    put_error(error_message);
+  
+  file.write(&output[0], output.size());
+}
+
+//=====make_key=====//
+void make_key(const std::string& filename)
+{
+  std::vector<Byte> input;
+  std::ifstream file (filename);
+  input.assign(std::istream_iterator<char>(file), std::istream_iterator<char>());
+  shuffle(input);
+  std::string keyname = filename + ".key";
+  write_file(keyname, input, "Couldn't open file to save the result.\n");
+}
+
+//=====read_key_position=====//
+std::streampos read_key_position(const std::string keyname)
+{
+  long tmp = 0;
+  std::ifstream meta (keyname + ".meta");
+
+  if (meta.is_open())
+    meta >> tmp;
+
+  return tmp;
+}
+
+//=====record_meta_to_both_files=====//
+void record_meta_to_both_files(const long key_starts, const long size_of_file, const std::string keyname, std::ofstream& out)
+{
+  std::ofstream meta_file (keyname + ".meta", std::ios::out|std::ios::trunc);
+  meta_file << key_starts + size_of_file;
+
+  std::string tmp = "<encrypted> meta=" + std::to_string(key_starts) + "\n";
```

Comment:
You can just out << tmp. But you have to be sure stream is in text mode. You can switch the mode any time using iomanips.

Rename tmp -> header. because it's a file header.
And maybe it would be better like <encrypted meta="..."/> I mean like XML?